### PR TITLE
jira-cli-go: init at 0.3.0

### PR DESCRIPTION
**Pull request by bryanasdev000**
###### Description of changes

Init jira-cli-go (alternative name, since a python version exists as `jira-cli`) at 0.3.0.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/jira-cli-go/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{ lib, buildGoModule, fetchFromGitHub, pkgs, installShellFiles }:
```
Comment (muscaln) [inline pkgs/development/tools/jira-cli-go/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, less, more, installShellFiles }:
```

File: pkgs/development/tools/jira-cli-go/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  vendorSha256 = "sha256-UO30/D65vpu3PgEsfSDL3nYgkwo5Cj+1WKiokk7KKKg=";
+
+  ldflags = [
+    "-s" "-w"
+    "-X github.com/ankitpokhrel/jira-cli/internal/version.GitCommit=${src.rev}"
+    "-X github.com/ankitpokhrel/jira-cli/internal/version.SourceDateEpoch=0"
+    "-X github.com/ankitpokhrel/jira-cli/internal/version.Version=${version}"
+  ];
+
+  checkInputs = with pkgs; [ less more ]; # Tests expect a pager in $PATH
```
Comment (muscaln) [inline pkgs/development/tools/jira-cli-go/default.nix:23]:
```suggestion
  checkInputs = [ less more ]; # Tests expect a pager in $PATH
```

**Review by muscaln — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/jira-cli-go/default.nix (hunk 1 above)
Comment (bryanasdev000) [reply pkgs/development/tools/jira-cli-go/default.nix]: Done, thanks!

**Comment by muscaln** [comment]
Result of `nixpkgs-review pr 173440` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jira-cli-go</li>
  </ul>
</details>

**Review by muscaln — APPROVED** [review]
LGTM

cc @SuperSandro2000 for review.

File: pkgs/development/tools/jira-cli-go/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "-X github.com/ankitpokhrel/jira-cli/internal/version.SourceDateEpoch=0"
+    "-X github.com/ankitpokhrel/jira-cli/internal/version.Version=${version}"
+  ];
+
+  checkInputs = [ less more ]; # Tests expect a pager in $PATH
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/jira version | grep ${version} > /dev/null
+  '';
```
Comment (thiagokokada) [inline pkgs/development/tools/jira-cli-go/default.nix:28]: You can use the `testVersion` instead.

File: pkgs/development/tools/jira-cli-go/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    $out/bin/jira completion bash > jira.bash
+    $out/bin/jira completion zsh > jira.zsh
+    installShellCompletion jira.{bash,zsh}
```
Comment (SuperSandro2000) [inline pkgs/development/tools/jira-cli-go/default.nix:34]:
```suggestion
    installShellCompletion --cmd jira \
      --bash <($out/bin/jira completion bash) \
      --zsh <($out/bin/jira completion zsh)
```

File: pkgs/development/tools/jira-cli-go/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
+{ lib, buildGoModule, fetchFromGitHub, less, more, installShellFiles }:
+
+buildGoModule rec {
+  pname = "jira-cli-go";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "ankitpokhrel";
+    repo = "jira-cli";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/jira-cli-go/default.nix:10]:
```suggestion
    owner = "ankitpokhrel";
    repo = "jira-cli";
    rev = "v${version}";
```

File: pkgs/development/tools/jira-cli-go/default.nix (hunk 3 above)
Comment (bryanasdev000) [reply pkgs/development/tools/jira-cli-go/default.nix:28]: Nice to know it :P will need to refactor my packages.

File: pkgs/development/tools/jira-cli-go/default.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    package = jira-cli-go;
+    command = "jira version";
+    inherit version;
+  };
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    installShellCompletion --cmd jira \
+    --bash <($out/bin/jira completion bash) \
+    --zsh <($out/bin/jira completion zsh)
```
Comment (SuperSandro2000) [inline pkgs/development/tools/jira-cli-go/default.nix:35]:
```suggestion
      --bash <($out/bin/jira completion bash) \
      --zsh <($out/bin/jira completion zsh)
```

**Comment by ankitpokhrel** [comment]
Whoa, thank you for working on this @bryanasdev000 🎉 

Would you mind adding installation instruction to the [jira-cli](https://github.com/ankitpokhrel/jira-cli) repo?

**Comment by bryanasdev000** [comment]
> Whoa, thank you for working on this @bryanasdev000 tada
> 
> Would you mind adding installation instruction to the [jira-cli](https://github.com/ankitpokhrel/jira-cli) repo?

Will do it soon :)

Thanks!

**Pull request merged**